.github/workflows/ci.yml

```
@@ -327,6 +325,26 @@ jobs:
           python make.py python-ci-install
         timeout-minutes: 20
 
+      # We only save the libparsec lib when:
+      # - We are not in a github queue branch (they're a one time use so caching won't help)
+      # - The rust code isn't modifed.
+      # - We haven't add a cache it.
+      - name: Save cached libparsec to be reused on later call
+        if: >-
+          steps.python-changes.outputs.run == 'true'
+          && steps.changes.outputs.rust-changes != 'true'
+          && steps.cache-libparsec.outputs.cache-hit != 'true'
+          && contains(github.ref, 'gh-readonly-queue') != 'true'
```

Wouldn't it be clearer to use `== 'false'` instead of `!= 'true'`?
```suggestion
          && steps.changes.outputs.rust-changes == 'false'
          && steps.cache-libparsec.outputs.cache-hit == 'false'
          && contains(github.ref, 'gh-readonly-queue') == 'false'
```

---

Yes, it would but it's more robust when the condition return `''` instead of `'false'`

---

indeed, the issue is it's very easy to break string comparison. So `!= 'true'` is less readable but save us from debugging headaches with the CI ^^